### PR TITLE
Restructured package.json and edited eslint config allowing using jsx.

### PR DESCRIPTION
**Pull request by fayzzzm**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,7 +38,19 @@
...
+    "node-sass": "^6.0.1",
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
+    "react-scripts": "4.0.3",
+    "web-vitals": "^1.1.2"
+  },
   "devDependencies": {
+    "@testing-library/jest-dom": "^5.14.1",
+    "@testing-library/react": "^11.2.7",
+    "@testing-library/user-event": "^12.8.3",
```
Comment (fayzzzm) [inline package.json:53]: Those things that we don't need in prod, should be moved to `devDependencies`. We don't want to have those modules in our chunk, so we could skip them, that's why I rewrote the code like this.

File: package.json [hunk 2]
```diff
@@ -50,7 +38,19 @@
...
   },
+  "dependencies": {
+    "@emmetio/codemirror-plugin": "^1.2.3",
+    "codemirror": "^5.60.0",
+    "node-sass": "^6.0.1",
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
+    "react-scripts": "4.0.3",
+    "web-vitals": "^1.1.2"
+  },
```
Comment (fayzzzm) [inline package.json:49]: moved `dependencies` and `devDependencies` to the bottom of `package.json` file, so we could at least, clarify the package structure of our application.

File: package.json [hunk 3]
```diff
@@ -3,18 +3,6 @@
...
-    "codemirror": "^5.60.0",
-    "node-sass": "^6.0.1",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2",
-    "react-scripts": "4.0.3",
-    "web-vitals": "^1.1.2"
-  },
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
```
Comment (fayzzzm) [inline package.json:8]: `scripts` section always should be at the top of our code. By doing this, we are making our task easier by finding which commands are available in our project.

File: .eslintrc.js [hunk 4]
```diff
@@ -2,12 +2,14 @@ module.exports = {
   extends: ['airbnb', 'prettier', 'plugin:jsx-a11y/recommended', 'plugin:react-hooks/recommended'],
   parser: 'babel-eslint',
   parserOptions: {
-    ecmaVersion: 8,
     ecmaFeatures: {
       experimentalObjectRestSpread: true,
       impliedStrict: true,
       classes: true,
+      jsx: true,
```
Comment (fayzzzm) [inline .eslintrc.js:9]: Allowing `jsx` parsing in order `linter`'s error appearance.

File: .eslintrc.js [hunk 5]
```diff
@@ -2,12 +2,14 @@ module.exports = {
...
     },
+    ecmaVersion: 12,
```
Comment (fayzzzm) [inline .eslintrc.js:11]: Allowing the latest features of javascript, at the top I replaced your version with mine.

**Comment by fayzzzm** [comment]
I am clarifying the way we are starting the project, so we could have at least an understanding of what / how we are doing the things in that project. Moving forward hearing your further steps about these things and what things should be done. 

If you have anything to add, please let me know.

File: package.json (hunk 3 above)
Comment (VoIdJMp) [reply package.json:8]: I agree with you

File: .eslintrc.js (hunk 5 above)
Comment (VoIdJMp) [reply .eslintrc.js:11]: Ок

**Review by VoIdJMp — APPROVED** [review]

**Pull request merged**